src/App.tsx

```diff
@@ -15,48 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 0,
+  Name = 1,
+  Length = 2,
+}
+
+type Filter = {
+  sortField: SortType,
+  isSorted: boolean,
+};
+
+function sortGoods(goods: string[], { sortField, isSorted }: Filter) {
+  const localGoods = [...goods];
+
+  if (sortField) {
+    localGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isSorted) {
+    localGoods.reverse();
+  }
+
+  return localGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [isSorted, setSorted] = useState(false);
```

Comment:
why do you need additional state check if sortField !== SortType.Default

Reply:
I need it for the handling reverse
renamed it for better understanding